src/Parsers/ASTFunction.cpp

```diff
@@ -324,10 +324,14 @@ void ASTFunction::formatImplWithoutAlias(const FormatSettings & settings, Format
 
             if (!written && 0 == strcmp(name.c_str(), "tupleElement"))
             {
-                /// It can be printed in a form of 'x.1' only if right hand side is unsigned integer literal.
+                // It can be printed in a form of 'x.1' only if right hand side
+                // is an unsigned integer lineral. We also allow nonnegative
+                // signed integer literals, because the fuzzer sometimes inserts
+                // them, and we want to have consistent formatting.
                 if (const auto * lit = arguments->children[1]->as<ASTLiteral>())
                 {
-                    if (lit->value.getType() == Field::Types::UInt64)
+                    if (isInt64FieldType(lit->value.getType())
```

Comment:
The name `isInt64FieldType` is non-obvious, I have to look into the source code.
Ok.

Reply:
You mean, it's called Int64, but also checks for UInt64? Not sure how to rename it, `is64BitIntFieldType` doesn't look much better...